src/classes/ClassWithStudentId.cls

```diff
@@ -0,0 +1,22 @@
+public with sharing class ClassWithStudentId{
+    /**
+    method that returns a map where key is class and value is set of studentid in it. 
+    */
+    public static Map<Class__c,Set<Student__c>> getClassesWithStudentId(){
+        List<Class__c> myClasses = [SELECT id,(SELECt id FROM Students__r) FROM Class__c]; 
```

Comment:
Write query in selector class only.

Reply:
You asked me to re write only the last two questions of this assignment using enterprise patterns. Do you want me to re write all of them?

Reply:
Yeah, I forgot.